pip/basecommand.py

```diff
@@ -313,6 +313,23 @@ def populate_requirement_set(requirement_set, args, options, finder,
                     'You must give at least one requirement to %(name)s '
                     '(see "pip help %(name)s")' % opts)
 
+        # On Windows, any operation modifying pip should be run as:
+        #     python -m pip ...
+        # See https://github.com/pypa/pip/issues/1299 for more discussion
+        should_show_use_python_msg = (
+            os.name == 'nt' and
```

Comment:
use ``sys.platform == 'win32'`` instead. Much cleaner and catches all Windows targets as well.